src/app/router.js

```diff
@@ -62,15 +64,38 @@
 
       activate();
 
+
       function activate() {
-        console.log('router Controller connected');
         self.closeSidenav();
+        if (_.isEqual(userService.user, userService.mockUser)) {
+          console.log('should execute getStorageData() only on page reload'); // require specs
+
+          getStorageData();
+          $timeout(function(){
+            getStorageData();
+            console.log('calculate executed');
+          }, 3000);
+        }
```

Comment:
That prevents against getting user data from localforage on change state.

getDefaultController() is activated each time state is changed. But I need to separate page load event from state change event.

- When user changes state userService.user is not empty (not.toEqual(userService.mockUser)) and "get from localforage" is not activated.
- When user loads the page for the first time, userService.user is empty and activate() triggers.

btw implementation changed https://github.com/VasylPosmit/health/blob/master/src/app/router.js#L74